SLAM initialization fails with AssertionError in default configuration

## Description

I encountered an issue when processing certain videos with the default configuration. The system fails during initialization with an assertion error indicating that `self.corr`, `self.inp`, and `self.f_net` are all `None`.

After investigation, I believe there might be a potential edge case in the `add_neighborhood_factors` method when both `seq_init=True` and `cross_view=True` are enabled.

## Expected Behavior
The system should successfully initialize and process videos regardless of the configuration combination.

## Actual Behavior
The system fails with the following error:
```
AssertionError: assert self.corr is not None and self.inp is not None and self.f_net is not None
```

## Root Cause Analysis
When `seq_init=True` and `cross_view=True`:
- `seq_init=True` sets `r=1` (radius)
- `cross_view=True` sets `c=1` (cross-view offset)

The filtering condition becomes:
```python
keep = ((ii - jj).abs() > c) & ((ii - jj).abs() <= r)
# Which translates to: abs(diff) > 1 AND abs(diff) <= 1
```

This condition appears to be mathematically impossible to satisfy, resulting in zero edges being added to the factor graph.

## Proposed Solution
```python
def add_neighborhood_factors(self, t0, t1, r: int = 3):
    ii, jj = torch.meshgrid(torch.arange(t0, t1), torch.arange(t0, t1), indexing="ij")
    ii = ii.reshape(-1).to(dtype=torch.long, device=self.device)
    jj = jj.reshape(-1).to(dtype=torch.long, device=self.device)

    c = 1 if self.cross_view else 0

    # Handle edge case when r <= c
    if r <= c:
        keep = ((ii - jj).abs() == r)
    else:
        keep = ((ii - jj).abs() > c) & ((ii - jj).abs() <= r)
    
    self.add_factors(ii[keep], jj[keep])
```

## Test Results
After applying this change, the previously failing video now processes successfully through the initialization phase.

## Questions
1. Is my understanding of the intended behavior correct?
2. Would this proposed solution be appropriate, or would you suggest a different approach?
3. Are there other configuration combinations that might have similar conflicts?

I'd be happy to submit a pull request with this fix if you think this approach is appropriate.

Thank you for your time and for maintaining this excellent project!

--- a/vipe/slam/components/factor_graph.py
+++ b/vipe/slam/components/factor_graph.py
@@ -403,7 +403,7 @@ class FactorGraph:
         ii = ii.reshape(-1).to(dtype=torch.long, device=self.device)
         jj = jj.reshape(-1).to(dtype=torch.long, device=self.device)
 
-        c = 1 if self.cross_view else 0
+        c = -1 if self.cross_view else 0
 
         keep = ((ii - jj).abs() > c) & ((ii - jj).abs() <= r)
         self.add_factors(ii[keep], jj[keep])

